### PR TITLE
Report: collapse aggregations when all audits pass

### PR DESCRIPTION
**Pull request by ebidel**
R: @patrickhulce @paulirish 

In the act of de-cluttering, this PR addresses some of of https://github.com/GoogleChrome/lighthouse/issues/1613 and implements the proposed solution in https://github.com/GoogleChrome/lighthouse/issues/1613#issuecomment-277126590.

Now, if all audits pass, the entire aggregation is collapsed. If one or more audits fail, the aggregation is opened up.

Before:

screen shot 2017-02-16 at 10 58 53 pm

After:

screen shot 2017-02-16 at 10 40 05 pm

when an audit fail: 

screen shot 2017-02-16 at 10 40 24 pm

BTW, the aggregation icons are slightly bigger than the subaudits. I thought the size difference was good but I'm happy to make them all the homogeneous. 

- Aggregations now always create an overall score. Please look over those changes as I may have missed something.
- Adds css transitions to the details dropdown.
- @patrickhulce turns out the `expectedValues`/`weights` are important and we'll need to use them after all :) aggregate.js uses them to type check and calculate an overall score based on the weights.

### CODE REVIEW

**Comment by patrickhulce** [comment]
> aggregate.js uses them to type check and calculate an overall score based on the weights

sure it uses them *now* ;) this is awesome though!

File: lighthouse-core/config/default.json [hunk 1]
```diff
@@ -387,11 +412,11 @@
           "weight": 1
         },
         "critical-request-chains": {
-          "expectedValue": 0,
```
Comment (patrickhulce) [inline lighthouse-core/config/default.json]: I thought this one actually was numeric, is it not?

**Comment by paulirish** [comment]
Totally on board with the collapsing. Nice.

Regarding scoring, are you saying we'll now have scores for "Best practices" and "fancier stuff"?

**Comment by ebidel** [comment]
> Regarding scoring, are you saying we'll now have scores for "Best practices" and "fancier stuff"?

Sorry. There's still only one score in the  report (PWA). The difference with this PR is that it calculates an overall score for each subitem. That's not surfaced in the report though.

Before (all `overall: 0` for unscored audits):

screen shot 2017-02-17 at 10 02 24 am

Now:

screen shot 2017-02-17 at 10 00 29 am

**Comment by patrickhulce** [comment]
After playing with it for a bit, I really like it, but the UX of the cards/boxes could use some work since I don't think it plays well with the collapsed report. Everything is rather succinct and cleanly laid out and the three large uncollapsible boxes really jump out. Can we have those collapse too?

image

**Comment by patrickhulce** [comment]
Yeah that's a fair point. Right now they just really stick out. If we used them more throughout I'd probably feel differently, but probably need to a strike a balance to prevent wall of text becoming wall of boxes :)

**Comment by paulirish** [comment]
> Personally I like seeing them b/c it's something other than text.

Agree with that, but if those numbers are in good shape then it's not adding much value to always show them.
(Also maybe the target for each value should be in the card, as I don't know what numbers are good for children and depth)

**Comment by ebidel** [comment]
Added target vals support to the cards and they're now hidden by default. PTAL.

screen shot 2017-02-17 at 12 40 26 pm

File: lighthouse-core/config/default.json (hunk 1 above)
Comment (ebidel) [inline lighthouse-core/config/default.json]: The `rawValue` that we display is numeric, yes. But we treat the audit as a pass/fail, which means `expectedValue` needs to be a boolean. AFAICT, `expectedValue` should really be `expectedType`.

**Comment by ebidel** [comment]
@patrickhulce you happy too?

**Comment by patrickhulce** [comment]
Oh sorry yep!

**Pull request merged**